src/main/java/org/andersen/homework/service/UserService.java

```diff
@@ -2,22 +2,30 @@
 
 import java.util.List;
 import java.util.UUID;
-import org.andersen.homework.model.dao.jdbc.JdbcDaoFactory;
+import lombok.RequiredArgsConstructor;
 import org.andersen.homework.model.dao.jdbc.UserDaoJdbc;
 import org.andersen.homework.model.entity.user.User;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
```

Comment:
What does do this annotation on class level?

Reply:
Applying `@Transactional(readOnly = true)` at the class level ensures that all methods in the class are treated as read-only by default, means that these methods are optimized for read operations and cannot modify db. For methods that should modify db I added `@Transactional` annotation to its signature to enables write operations for them while keeping the rest of the class read-only.